### PR TITLE
Log error when default profiles xml loads with error + unittest [5596]

### PR DESCRIPTION
**Pull request by EduPonz**

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1857/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1816/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1828/)

File: src/cpp/xmlparser/XMLParser.cpp [hunk 1]
```diff
@@ -2540,8 +2540,10 @@ XMLP_ret XMLParser::loadXML(const std::string& filename, up_base_node_t& root)
     }
 
     tinyxml2::XMLDocument xmlDoc;
-    if (tinyxml2::XMLError::XML_SUCCESS != xmlDoc.LoadFile(filename.c_str()))
+    tinyxml2::XMLError load_error = xmlDoc.LoadFile(filename.c_str());
+    if (tinyxml2::XMLError::XML_SUCCESS != load_error)
     {
+        logError(XMLPARSER, "Error loading '" << filename << "'. Error code: " << load_error);
```
Comment (raquelalvarezbanos) [inline src/cpp/xmlparser/XMLParser.cpp]: Perhaps you could add more info here and say that next we'll try to load the default profile

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1881/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1840/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1852/)

File: src/cpp/xmlparser/XMLParser.cpp [hunk 2]
```diff
@@ -2540,11 +2540,10 @@ XMLP_ret XMLParser::loadXML(const std::string& filename, up_base_node_t& root)
     }
 
     tinyxml2::XMLDocument xmlDoc;
-    if (tinyxml2::XMLError::XML_SUCCESS != xmlDoc.LoadFile(filename.c_str()))
+    tinyxml2::XMLError load_error = xmlDoc.LoadFile(filename.c_str());
+    if (tinyxml2::XMLError::XML_SUCCESS != load_error)
```
Comment (raquelalvarezbanos) [inline src/cpp/xmlparser/XMLParser.cpp]: Can we combine the two lines above into a single one?

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1905/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1864/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1877/)

**Pull request merged**